catalog/spatial/registry/registry-api-impl/src/main/java/org/codice/ddf/registry/api/impl/RegistryStoreImpl.java

```diff
@@ -246,10 +247,10 @@ public SourceResponse query(QueryRequest request) throws UnsupportedQueryExcepti
 
         SourceResponse registryQueryResponse = super.query(request);
         for (Result singleResult : registryQueryResponse.getResults()) {
-            if (singleResult.getMetacard()
-                    .getAttribute(RegistryObjectMetacardType.REGISTRY_ID)
-                    .getValue()
-                    .toString()
+            if (!RegistryUtility.isRegistryMetacard(singleResult.getMetacard())) {
```

Comment:
We weren't, but we should have been checking for the id, isRegistryMetacard can do that for us. If we don't have the tag, we shouldn't have the id. Someone is using the system improperly if this occurs. This if  statement prevents the need for a null check in the next comment since this will break if the id attribute is null as well.